src/preprocessor.ts

```diff
@@ -21,7 +24,10 @@ export function preprocessor(code: string, options: PrettierParserOptions) {
 
     traverse(ast, {
         ImportDeclaration(path: NodePath<ImportDeclaration>) {
-            importNodes.push(path.node);
+            const tsModuleParent = path.findParent((p) => isTSModuleDeclaration(p));
```

Comment:
can we check this after traversing AST ?  May be we can short-circuit for the *.d.ts files

Reply:
I'm not really sure that we can do that 🤔

Reply:
Also, do you see any other case .d.ts files that this change might break it?